test/phase_info_test.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import openhtf
+
+
+def PlainFunc():
+  """Plain Docstring"""
+  pass
+
+
+def ExtraArgFunc(input=None):
+  return input
+
+
+class TestPhaseInfo(unittest.TestCase):
+
+  def testBasics(self):
+      phase = openhtf.PhaseInfo.WrapOrCopy(PlainFunc)
+      assert phase.func is PlainFunc
+      assert len(phase.plugs) == 0
+      assert phase.name == 'PlainFunc'
+      assert phase.doc == 'Plain Docstring'
+      phase(None)
+
+  def testMultiplePhases(self):
+      # Tests if you can call WrapOrCopy on a PhaseInfo object and modify the
+      # result without modifying the original.
+      phase = openhtf.PhaseInfo.WrapOrCopy(PlainFunc)
+      phase.plugs.append(1)
+      assert len(phase.plugs) == 1
+
+      second_phase = openhtf.PhaseInfo.WrapOrCopy(phase)
+      second_phase.plugs.append(2)
+      assert len(phase.plugs) == 1
+      assert len(second_phase.plugs) == 2
+
+  def testWithArgs(self):
+      phase = openhtf.PhaseInfo.WrapOrCopy(ExtraArgFunc)
+      phase = phase.WithArgs(input='input arg')
```

Comment:
instead of reassigning to phase, you might save the original and do `assertIsNot` checks between the three.  it's a little redundant with the above test, so i wouldn't bother checking attributes as well, but a good sanity check that `.WithArgs()` is making new things instead of modifying the original.